_posts/en/newsletters/2019-05-28-newsletter.md

```diff
@@ -0,0 +1,597 @@
+to sign this distribution transaction and she ensures each person
+receives everyone else's signatures, then she signs and broadcasts the
+setup transaction.
+
+![FIXME](/img/posts/2019-05-tx-tree-1.png)
+
+When the setup transaction receives a reasonable number of
+confirmations, there's no way for Alice to take back her payment to the
+10 receivers.  As long as each of the receivers has a copy of the
+distribution transaction and all the others' signatures, there's also no
+way for any receivers to cheat any other receiver out of a payment.  So
+even though the distribution transaction that actually pays the
+receivers hasn't been broadcast or confirmed, the payments have the same
+amount of security as the confirmed setup transaction.  At any time, any
+of the receivers who wants to spend their money further can broadcast
+the distribution transaction and wait for it to confirm (perhaps helping
+to pay its fee using child-pays-for-parent fee bumping).
+
+This technique allows spenders and receivers to lock in a set of
+payments during high fees and then only distribute the actual payments when
+fees are lower.  According to Bitcoin Core fee estimates at the time of
+writing, anyone patient enough to wait a week for a transaction to
+confirm (like the distribution transaction above) can save 99% on fees.
+Let's look at the example above in that context.  To make later
+comparisons to Taproot more fair, we'll assume some form of key and
+signature aggregation is being used, such as [MuSig][] or (in theory)
+multiparty ECDSA (see [Newsletter #18][]).
+
+| | Individual Payments | Batched Payment | Commit now, distribute later |
+|-|-|-|-|
+| Immediate (high fee) transactions | 10x141 vbytes | 1x411 vbytes | 1x141 vbytes |
+| Cost at 0.00142112 BTC/KvB | 0.00204641 | 0.00058409 | 0.0002003 |
+| Delayed (low fee) transactions | --- | --- | 1x381 vbytes |
+| Cost at 0.00001014 BTC/KvB | --- | --- | 0.00000386 |
+| Total vbytes | 1,410 | 411 | 522 |
+| Total cost | 0.00204641 | 0.00058409 | 0.00020416 |
+| **Savings compared to previous column** | --- | **71%** | **65%** |
+
+We see that this type of trustlessly delayed payment can save 65% over
+payment batching and 90% over sending separate payments.  Note that the
+savings could be even larger during periods of greater fee
+stratification or with more than ten receivers.
+
+### CheckOutputsHashVerify
+
+The proposed soft fork would add a new opcode,
+`OP_CHECKOUTPUTSHASHVERIFY` (abbreviated by its author as `OP_COSHV`
+with an extra *S*).  This opcode and a hash digest could be included in
+tapleaf scripts, allowing it to be one of the conditions in a Taproot
+address.  When that address was spent, if COSHV was executed, the
+spending transaction would only be valid if the hash digest of its
+outputs matched the hash digest read from the script by COSHV.
+
+Comparing this to our example above, Alice would again ask each of the
+participants for a public key (such as a Taproot
```

Comment:
This isn't necessary -- Alice an instead generate a NUMS point P (that can't be used for signing), generate the OP_COSHV script S, calculate the taproot point Q = P+H(P,S)*G, and pay to Q. To spend, you need to know the NUMS point P, the script structure for S and the outputs that are being paid to.

The advantage of having P as a musig point instead is the recipients can then change the arrangement later if they want.

Reply:
@ajtowns

> This [Alice asking participants for their pubkeys] isn't necessary

It is, because that pubkey (as a Taproot address) is where the output will ultimately be sent.  I say "pubkey" here instead of "address" so that this looks almost identical to the example in the previous section using current script.  There's a parenthetical here saying the pubkey is the taproot address and a linked footnote about why you shouldn't assume taproot addresses are necessarily pubkeys.

> The advantage of having P as a musig point instead is the recipients can then change the arrangement later if they want.

Since we're going to use that feature in later subsections, I think it's best that we introduce Alice here as using musig to create the taproot internal key.  I think it's always better for her to do so anyway.